### PR TITLE
Update Inner Blocks (ESNext) to use latest API

### PR DESCRIPTION
**Pull request by ryanwelcher**
Part of the effort in #152

### CODE REVIEW

**Review by gziolo — APPROVED** [review]
It's great to see all those updates 👍🏻

File: 06-inner-blocks-esnext/src/index.js [hunk 1]
```diff
@@ -1,21 +1,19 @@
...
+ * Internal dependencies
+ */
+import json from '../block.json';
+import edit from './edit';
+import save from './save';
+
+const { name, ...settings } = json;
+
+registerBlockType( name, {
+	...settings,
```
Comment (gziolo) [inline 06-inner-blocks-esnext/src/index.js:16]: One note. Technically speaking, you don't need to spread settings here. They are exposed from the server when registered with `register_block_type`.

Comment (ryanwelcher) [reply 06-inner-blocks-esnext/src/index.js:16]: Thanks for confirming that! It was a question that came up in a couple of live streams.

Comment (gziolo) [reply 06-inner-blocks-esnext/src/index.js:16]: https://github.com/WordPress/gutenberg/blob/1a3bda10d3d04233e864d89eb9db3ca2adb0e62c/packages/blocks/src/api/registration.js#L258-L271

That means that whatever the client provides overrides what the server provides (if registered also on the server).

If you register blocks only with JavaScript then you surely need to spread settings from `block.json` here.

Comment (gziolo) [reply 06-inner-blocks-esnext/src/index.js:16]: Example for core blocks:

https://github.com/WordPress/gutenberg/blob/1a3bda10d3d04233e864d89eb9db3ca2adb0e62c/packages/block-library/src/paragraph/index.js#L25-L57

It's only settings specific to the client (UI).

**Pull request merged**